### PR TITLE
Callum style layout improvements and my layout changes

### PR DESCRIPTION
**Pull request by daliusd**
## Description

This changeset mainly includes updated Callum style layout implementation:
* Simplified configuration for Callum style layout
* Added documentation how to use it if you want to use it with your keyboard
* Added unit-tests in case you want to change something

As well I have change my layout. There are minor things that some people might find interesting.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/a_dux/keymaps/daliusd/tests/minunit.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
+#define mu_assert(message, test) \
```
Comment (drashna) [inline keyboards/a_dux/keymaps/daliusd/tests/minunit.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/a_dux/keymaps/daliusd/tests/test.c [hunk 2]
```diff
@@ -0,0 +1,828 @@
+#include <stdint.h>
```
Comment (drashna) [inline keyboards/a_dux/keymaps/daliusd/tests/test.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/a_dux/keymaps/daliusd/tests/minunit.h (hunk 1 above)
Comment (daliusd) [reply keyboards/a_dux/keymaps/daliusd/tests/minunit.h:1]: Done.

File: keyboards/a_dux/keymaps/daliusd/tests/test.c (hunk 2 above)
Comment (daliusd) [reply keyboards/a_dux/keymaps/daliusd/tests/test.c:17]: Done.

File: keyboards/a_dux/keymaps/daliusd/tests/minunit.h [hunk 3]
```diff
@@ -0,0 +1,90 @@
+/*
```
[the remark this answers is not in the document]
Comment (daliusd) [reply keyboards/a_dux/keymaps/daliusd/tests/minunit.h:1]: Well, that was easiest way to add unit-tests in user space. What alternatives would you suggest?

[the remark this answers is not in the document]
Comment (daliusd) [reply keyboards/a_dux/keymaps/daliusd/tests/minunit.h:1]: What's the decision here? BTW if that helps the case this wouldn't be first `minunit.h` in this repo: https://github.com/qmk/qmk_firmware/blob/9dea6f772077dc5c09daf40378e45884d29ab2e2/users/dennytom/chording_engine/tests/minunit.h

[the remark this answers is not in the document]
Comment (daliusd) [reply keyboards/a_dux/keymaps/daliusd/tests/minunit.h:1]: Hey, what changes are we waiting here?

[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/a_dux/keymaps/daliusd/tests/minunit.h:1]: I think for now, the best course of action is to remove the tests.

[the remark this answers is not in the document]
Comment (daliusd) [reply keyboards/a_dux/keymaps/daliusd/tests/minunit.h]: I have removed tests.

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by daliusd** [comment]
@drashna thanks

**Comment by Abraxas-Knister** [comment]
this was merged, why can't I find it in the current master? Is is still expected to work?

**Comment by fauxpark** [comment]
https://github.com/qmk/qmk_firmware/issues/22724

**Pull request merged**